src/metatensor/models/cli/train_model.py

```diff
@@ -87,13 +188,82 @@ def train_model(options: DictConfig) -> None:
+        filename=conf_training_set["structures"]["read_from"],
+        fileformat=conf_training_set["structures"]["file_format"],
     )
-    dataset = Dataset(structures, targets)
+    targets_train = read_targets(conf_training_set["targets"])
+    train_dataset = Dataset(structures_train, targets_train)
+
+    logger.info("Setting up test set")
+    conf_test_set = options["test_set"]
+    if not isinstance(conf_test_set, float):
+        conf_test_set = expand_dataset_config(conf_test_set)
+        structures_test = read_structures(
+            filename=conf_training_set["structures"]["read_from"],
+            fileformat=conf_training_set["structures"]["file_format"],
+        )
+        targets_test = read_targets(conf_test_set["targets"])
+        test_dataset = Dataset(structures_test, targets_test)
+        fraction_test_set = 0.0
+    else:
+        if conf_test_set < 0 or conf_test_set >= 1:
+            raise ValueError("Test set split must be between 0 and 1.")
+        fraction_test_set = conf_test_set
+
+    logger.info("Setting up validation set")
+    conf_validation_set = options["validation_set"]
+    if not isinstance(conf_validation_set, float):
+        conf_validation_set = expand_dataset_config(conf_validation_set)
+        structures_validation = read_structures(
+            filename=conf_training_set["structures"]["read_from"],
+            fileformat=conf_training_set["structures"]["file_format"],
+        )
+        targets_validation = read_targets(conf_validation_set["targets"])
+        validation_dataset = Dataset(structures_validation, targets_validation)
+        fraction_validation_set = 0.0
+    else:
+        if conf_validation_set < 0 or conf_validation_set >= 1:
+            raise ValueError("Validation set split must be between 0 and 1.")
+        fraction_validation_set = conf_validation_set
+
+    # Split train dataset if requested
+    if fraction_test_set or fraction_validation_set:
+        fraction_train_set = 1 - fraction_test_set - fraction_validation_set
+        if fraction_train_set < 0:
+            raise ValueError("fraction of the train set is smaller then 0!")
+
+        # ignore warning of possible empty dataset
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            subsets = torch.utils.data.random_split(
+                dataset=train_dataset,
+                lengths=[
+                    fraction_train_set,
+                    fraction_test_set,
+                    fraction_validation_set,
+                ],
+                generator=generator,
+            )
+
+        train_dataset = subsets[0]
+        if fraction_test_set and not fraction_validation_set:
+            test_dataset = subsets[1]
+        elif not fraction_validation_set and fraction_validation_set:
+            validation_dataset = subsets[1]
+        else:
+            test_dataset = subsets[1]
+            validation_dataset = subsets[2]
```

Comment:
I'm too lazy to think about the code, what happens here if the user specifies a validation set but then sets a test set of 0.1?

Reply:
then 90% of the original train set goes into training and 10% of the original training set into validation. Does this makes sense for you?

Reply:
Yes, perfect